common/pubsub/pubsub_async.go

```diff
@@ -0,0 +1,49 @@
+package pubsub
+
+import (
+	"context"
+
+	"github.com/status-im/status-go/common"
+)
+
+// PublishAsync will send the value val of type T on the specified Publisher in a
+// non-blocking manner.
+// If the context is canceled, the value may not be sent to all subscribers.
+func PublishAsync[T any](ctx context.Context, p *Publisher, val T) {
+	go func() {
+		defer common.LogOnPanic()
+		PublishSync(ctx, p, val)
+	}()
+}
+
+// SubscribeAsync creates a channel to listen for events of type T published by the Publisher.
+// The returned channel will be closed by the Publisher.
+// UnsubFn must be called by the subscriber before it becomes unable to process events. It will
+// return immediately, potentially missing some events.
+func SubscribeAsync[T any](p *Publisher) (chan T, UnsubFn) {
+	subs := getTypePublisher[T](p)
+	ch := subs.Subscribe()
+
+	unsub := func() {
+		subs.Unsubscribe(ch)
+	}
+
+	return ch, unsub
+}
+
+// SubscribeCallbackAsync connects a callback to events of type T published by the Publisher.
+// UnsubFn must be called by the subscriber before it becomes unable to process events. It will
+// return immediately, potentially missing some events.
+func SubscribeAsyncCallback[T any](p *Publisher, callback func(T)) UnsubFn {
```

Comment:
hmmm, I don't quite like it. Such wrappers can be introduced as needed by the client, but I wouldn't encourage it by adding this to the API. I'd keep explicit usage of channels.

But I don't have a strong opinion, if you think it's needed.

Reply:
> ... listen to a channel and call a function when an event was received

Yeah that's true, you always do some action based on a channel message.

I just think it's better to keep it Go-style channels, rather than JS-style callbacks